client/my-sites/checkout/index.js

```diff
@@ -199,7 +218,8 @@ export default function () {
 	);
 
 	page(
-		'/checkout/:domainOrProduct',
```

Comment:
With this removal, I think that the route `/checkout/:productSlug` will no longer work correctly.